test/e2e/apimachinery/custom_resource_definition.go

```diff
@@ -42,29 +51,153 @@ var _ = SIGDescribe("CustomResourceDefinition resources", func() {
+			}()
+		})
+
+		/*
+			Release : v1.16
+			Testname: Custom Resource Definition, list
+			Description: Create a API extension client, define 10 random custom resource definitions and list them using a label selector. API server MUST be able to list the custom resource definitions and delete them via delete collection.
+		*/
+		ginkgo.It("listing custom resource definition objects works ", func() {
+			testListSize := 10
+			config, err := framework.LoadConfig()
+			framework.ExpectNoError(err, "loading config")
+			apiExtensionClient, err := clientset.NewForConfig(config)
+			framework.ExpectNoError(err, "initializing apiExtensionClient")
+
+			// Label the CRDs we create so we can list only them even though they are cluster scoped
+			testUUID := string(uuid.NewUUID())
+
+			// Create CRD and wait for the resource to be recognized and available.
+			crds := make([]*v1beta1.CustomResourceDefinition, testListSize)
+			for i := 0; i < testListSize; i++ {
+				crd := fixtures.NewRandomNameCustomResourceDefinition(v1beta1.ClusterScoped)
+				crd.Labels = map[string]string{"e2e-list-test-uuid": testUUID}
+				crd, err = fixtures.CreateNewCustomResourceDefinition(crd, apiExtensionClient, f.DynamicClient)
+				framework.ExpectNoError(err, "creating CustomResourceDefinition")
+				crds[i] = crd
+			}
+
+			// Create a crd w/o the label to ensure the label selector matching works correctly
+			crd := fixtures.NewRandomNameCustomResourceDefinition(v1beta1.ClusterScoped)
+			crd, err = fixtures.CreateNewCustomResourceDefinition(crd, apiExtensionClient, f.DynamicClient)
+			framework.ExpectNoError(err, "creating CustomResourceDefinition")
+			defer func() {
+				err = fixtures.DeleteCustomResourceDefinition(crd, apiExtensionClient)
+				framework.ExpectNoError(err, "deleting CustomResourceDefinition")
+			}()
+
+			selectorListOpts := metav1.ListOptions{LabelSelector: "e2e-list-test-uuid=" + testUUID}
+			list, err := apiExtensionClient.ApiextensionsV1beta1().CustomResourceDefinitions().List(selectorListOpts)
+			framework.ExpectNoError(err, "listing CustomResourceDefinitions")
+			gomega.Expect(len(list.Items)).To(gomega.Equal(testListSize))
+			for _, actual := range list.Items {
+				var expected *v1beta1.CustomResourceDefinition
+				for _, e := range crds {
+					if e.Name == actual.Name && e.Namespace == actual.Namespace {
+						expected = e
+					}
+				}
+				gomega.Expect(expected).To(gomega.Not(gomega.BeNil()))
+				if !equality.Semantic.DeepEqual(actual.Spec, expected.Spec) {
+					e2elog.Failf("Expected CustomResourceDefinition in list with name %s to match crd created with same name, but got different specs:\n%s",
+						actual.Name, diff.ObjectReflectDiff(expected.Spec, actual.Spec))
 				}
+			}
+
+			// Use delete collection to remove the CRDs
+			err = fixtures.DeleteCustomResourceDefinitions(selectorListOpts, apiExtensionClient)
+			framework.ExpectNoError(err, "deleting CustomResourceDefinitions")
```

Comment:
after calling deletecollection with the label selector, verify `crd` (unlabeled CRD) still exists?